test/suites/test_pool.py

```diff
@@ -0,0 +1,357 @@
+
+        self.assertSetEqual(RW_ports_result, RW_ports)
+
+        # Expect PREFER_RW iterate through all RW instances if there is at least one.
+        PREFER_RW_ports_result = set()
+        for i in range(len(self.servers)):
+            resp = self.pool.eval('return box.cfg.listen', mode=tarantool.Mode.PREFER_RW)
+            PREFER_RW_ports_result.add(resp.data[0])
+
+        self.assertSetEqual(PREFER_RW_ports_result, RW_ports)
+
+        # Expect PREFER_RO iterate through all RO instances if there is at least one.
+        PREFER_RO_ports_result = set()
+        for i in range(len(self.servers)):
+            resp = self.pool.eval('return box.cfg.listen', mode=tarantool.Mode.PREFER_RO)
+            PREFER_RO_ports_result.add(resp.data[0])
+
+        self.assertSetEqual(PREFER_RO_ports_result, RO_ports)
+
+        # Expect PREFER_RW iterate through all instances if there are no RW.
+        self.set_cluster_ro([True, True, True, True, True])
+
+        def expect_PREFER_RW_iterate_through_all_instances_if_there_are_no_RW():
+            PREFER_RW_ports_result_all_ro = set()
+            for i in range(len(self.servers)):
+                resp = self.pool.eval('return box.cfg.listen', mode=tarantool.Mode.PREFER_RW)
+                PREFER_RW_ports_result_all_ro.add(resp.data[0])
+
+            self.assertSetEqual(PREFER_RW_ports_result_all_ro, all_ports)
+    
+        self.retry(func=expect_PREFER_RW_iterate_through_all_instances_if_there_are_no_RW)
+
+        # Expect PREFER_RO iterate through all instances if there are no RO.
+        self.set_cluster_ro([False, False, False, False, False])
+
+        def expect_PREFER_RO_iterate_through_all_instances_if_there_are_no_RO():
+            PREFER_RO_ports_result_all_rw = set()
+            for i in range(len(self.servers)):
+                resp = self.pool.eval('return box.cfg.listen', mode=tarantool.Mode.PREFER_RO)
+                PREFER_RO_ports_result_all_rw.add(resp.data[0])
+
+            self.assertSetEqual(PREFER_RO_ports_result_all_rw, all_ports)
+
+        self.retry(func=expect_PREFER_RO_iterate_through_all_instances_if_there_are_no_RO)
+
+    def test_02_exception_raise(self):
+        self.set_cluster_ro([False, False, True, False, True])
+
+        self.pool = tarantool.ConnectionPool(addrs=self.addrs, user='test', password='test')
+        with self.assertRaises(DatabaseError):
+            self.pool.call('non_existing_procedure')
+
+    def test_03_insert(self):
+        self.set_cluster_ro([False, True, False, True, True])
+        self.pool = tarantool.ConnectionPool(addrs=self.addrs, user='test', password='test')
+
+        self.assertSequenceEqual(self.pool.insert('test', ['test_05_insert_1', 1]), [['test_05_insert_1', 1]])
+        self.assertSequenceEqual(self.pool.insert('test', ['test_05_insert_2', 2]), [['test_05_insert_2', 2]])
+        self.assertSequenceEqual(self.pool.insert('test', ['test_05_insert_3', 3]), [['test_05_insert_3', 3]])
```

Comment:
The initial thought was to testinsert + roundrobin, but since it is not a popular case and all other RW-tests use a single master, I'll simplify that one.